src/libraries/Common/src/Interop/Windows/WinSock/Interop.WSADuplicateSocket.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Net.Sockets;
+using System.Runtime.InteropServices;
+
+internal static partial class Interop
+{
+    internal static partial class Winsock
+    {
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct WSAProtocolChain
+        {
+            internal int ChainLen;
+            [MarshalAs(UnmanagedType.ByValArray, SizeConst=7)]
+            internal uint[] ChainEntries;
```

Comment:
Rather than allocating an actual array, can you just have this directly mirror the native definition by using a fixed buffer?
```C#
private const int MAX_PROTOCOL_CHAIN = ...;
...
internal fixed uint ChainEntries[MAX_PROTOCOL_CHAIN];
```

Reply:
I did not bother turning stuff blittable (just reused FW code), since socket duplication doesn't look like something happening on hot application path. Will fix it.